packages/flutter_tools/test/general.shard/common_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../src/common.dart';
+
+void main() {
+  final Map<dynamic, dynamic> stubMatchState = <dynamic, dynamic>{};
+  group('containsIgnoreWhitespace Matcher', () {
+    group('on item to be contained', () {
+      test('matches simple case.', () {
+        final Matcher testMatcher = containsIgnoreWhitespace('any text!');
+        final bool result =
+            testMatcher.matches('Give me any text!', stubMatchState);
+        expect(result, true);
+      });
+
+      test("shouldn't match when it's only because of removing whitespaces", () {
```

Comment:
This test result is a little surprising to me. Could you clarify the reason why we wouldn't want `any text` to match `Give me anytext`? And if so, we should probably document in the code.

Reply:
I checked documentation for "equalsIgnoreWhitespace" and it won't match if "hello world" should be matched against "This is a helloworld example.". Actually it's hard to tell for me what should be the correct behaviour.It would be nice if you can give me any hints here:
https://api.flutter.dev/flutter/package-matcher_matcher/equalsIgnoringWhitespace.html

Reply:
Ahhh, I see. If I'm reading that right, what the linked matcher is doing is:

1) trimming leading and trailing whitespace
2) normalizing all whitespace to spaces
3) collapsing all repeated whitespace to a single space

Which makes sense to me.

Reply:
Should I then implement similar logic here? What you've described sounds very reasonable.

Reply:
I reimplemented the logic using "colapseWhitespace" function.